### ISSUE TITLE
Binary classification task in TFT model that outputs a probability distribution for next 10 time steps

### ISSUE THREAD

**Issue by Kimonili** [jdb78/pytorch-forecasting#622]
Hi there,

1. I have a binary classification task and I am using the TFT model. I want to output the probability of one of the two possible outcomes (binary) for each of the next 10 time steps. Is that possible with TFT?

2. How can we add a metric (accuracy, f1 score, precision, recall etc) to visualize the performance of the TFT model in classification tasks? Is it possible through the class directly or should I inherit it?

3. If I want 1 probability output for each of the next 10 timesteps given the last 30 timesteps, should I set `decoder_length=10`, `encoder_length=30` and `output_size=1` and the model will output the desired results or there is more to do to achieve that?

Thanks @jdb78 !

**Comment by Kimonili** [comment]
Any updates on this issue @jdb78?

**Comment by ammargilani** [comment]
Hi, any updates? @jdb78?

**Comment by Nintorac** [comment]
for 2 checkout `logging_metrics` which is a kwarg to `TemporalFusionTransformer`.

`output_size` I am also a little lost on but I think it's changing the number of quantiles for the quantile loss ([section 5](https://arxiv.org/abs/1912.09363)). Not what you are trying to do.

I just replaced my bool column with a category column with two categories, `true` and `false` and then used `pytorch_forecasting.metrics.CrossEntropy`, seems to work but please correct me if I've made a mistake
